concrete-integer/src/server_key/add_parallel.rs

```diff
@@ -0,0 +1,376 @@
+use crate::server_key::CheckError;
+use std::sync::{Condvar, Mutex};
+
+use crate::ciphertext::Ciphertext;
+use crate::ServerKey;
+
+impl ServerKey {
+    /// Computes homomorphically an addition between two ciphertexts encrypting integer values.
+    ///
+    /// This function computes the operation without checking if it exceeds the capacity of the
+    /// ciphertext.
+    ///
+    /// The result is returned as a new ciphertext.
+    ///
+    /// # Warning
+    ///
+    /// - Multithreaded
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use concrete_integer::gen_keys;
+    /// use concrete_shortint::parameters::PARAM_MESSAGE_2_CARRY_2;
+    ///
+    /// let size = 4;
+    ///
+    /// // Generate the client key and the server key:
+    /// let (cks, sks) = gen_keys(&PARAM_MESSAGE_2_CARRY_2, size);
+    ///
+    /// let msg1 = 10;
+    /// let msg2 = 127;
+    ///
+    /// let ct1 = cks.encrypt(msg1);
+    /// let ct2 = cks.encrypt(msg2);
+    ///
+    /// // Compute homomorphically an addition:
+    /// let ct_res = sks.unchecked_add_parallelized(&ct1, &ct2);
+    ///
+    /// // Decrypt:
+    /// let dec_result = cks.decrypt(&ct_res);
+    /// assert_eq!(dec_result, msg1 + msg2);
+    /// ```
+    pub fn unchecked_add_parallelized(
+        &self,
+        ct_left: &Ciphertext,
+        ct_right: &Ciphertext,
+    ) -> Ciphertext {
+        self.unchecked_add(ct_left, ct_right)
+    }
```

Comment:
my thought process was that this would make it easier to separate the engines into a parallel one and a sequential one, by having separate functions for each one. they could later be modified to have parallelism in their implementation (which would require changes in shortint/core first)

Reply:
The things that I don't particularly fancy is that when looking at this PR, is that withing  some `*_parallelized` function implementation we sometimes call another `_parallelized` function and sometimes not (eg in `smart_add_parallelized` and `smart_add_assign_parallelized`)

And more importantly, it means that when I see a `*_parallelized` function being called, I can't be sure its actually parallelized unless I go look how its implemented (which may require do go "deep")